.github/workflows/main.yaml

```diff
@@ -239,6 +239,12 @@ jobs:
           # Use fetch depth of 0 to get full history for a valid build id.
           fetch-depth: 0
           persist-credentials: false
+      - name: Setup Gradle
```

Comment:
@oxve Have a look here please. It's cheap and grungy way of running gradlew install right at the start with a few retries. Only the first cycle through the loop actually downloads and installs gradle, the remaining ones execute very quickly as it's already locally there.

The only negative side effect is a bit more log spam. And this only runs for Android.

Reply:
Very elegant